ui/src/app/dashboard/controllers/DashCtrl.js

```diff
@@ -2,17 +2,15 @@
 
   angular
     .module('app')
-    .controller('DashCtrl', [ '$scope', 'ToastService', 'DashService',
-      DashCtrl
-    ]);
+    .controller('DashCtrl', [ '$scope', 'ToastService', 'DashService', 'UserService', DashCtrl]);
 
-  function DashCtrl($scope, ToastService, DashService) {
+  function DashCtrl($scope, ToastService, DashService, UserService) {
     $scope.dlink = {link: ''};
     $scope.downloads = [];
-    $scope.username = localStorage.getItem('Username');
+    $scope.username = UserService.getUsername();
+    var socket;
```

Comment:
Here what I meant was to have it like this,

`
      var socket = io.connect('http://localhost:5000/progress');
      $scope.dlink = {link: ''};
      $scope.downloads = [];
      //rest of the code.
`